src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -100,17 +100,19 @@ trait AttributeComponent {
       filter(_.id inSetBind attributeRecords.map(_.id)).delete
     }
 
-    def unmarshalAttributes(allAttributeRecsWithRef: Seq[(AttributeRecord, Option[EntityRecord])]): Map[String, Attribute] = {
-      allAttributeRecsWithRef.groupBy(_._1.name).map { case (name, attributeRecsWithRefForNameWithDupes) =>
-        val attributeRecsWithRefForName = attributeRecsWithRefForNameWithDupes.toSet
-        if (attributeRecsWithRefForName.forall(_._1.listIndex.isDefined)) {
-          name -> unmarshalList(attributeRecsWithRefForName)
-        } else if (attributeRecsWithRefForName.size > 1) {
-          throw new RawlsException(s"more than one value exists for attribute $name but list index is not defined for all, records: $attributeRecsWithRefForName")
-        } else if (attributeRecsWithRefForName.head._2.isDefined) {
-          name -> unmarshalReference(attributeRecsWithRefForName.head._2.get)
-        } else {
-          name -> unmarshalValue(attributeRecsWithRefForName.head._1)
+    def unmarshalAttributes[ID](allAttributeRecsWithRef: Seq[((ID, AttributeRecord), Option[EntityRecord])]): Map[ID, Map[String, Attribute]] = {
```

Comment:
old version assumed all records belong to same entity, new version groups by an ID